komga/src/main/kotlin/org/gotson/komga/interfaces/rest/SeriesController.kt

```diff
@@ -324,12 +328,52 @@ class SeriesController(
     @AuthenticationPrincipal principal: KomgaPrincipal,
     @PathVariable(name = "seriesId") seriesId: String
   ): ByteArray {
-    seriesRepository.getLibraryId(seriesId)?.let {
-      if (!principal.user.canAccessLibrary(it)) throw ResponseStatusException(HttpStatus.FORBIDDEN)
-    } ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    val series = seriesRepository.getLibraryId(seriesId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    if (!principal.user.canAccessLibrary(series)) throw ResponseStatusException(HttpStatus.FORBIDDEN)
 
-    return seriesLifecycle.getThumbnailBytes(seriesId, principal.user.id)
-      ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    val thumbnail = seriesLifecycle.getThumbnailBytes(seriesId, principal.user.id)
+    return thumbnail ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+  }
+
+  @PatchMapping(value = ["{seriesId}/thumbnail"], consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
+  @PreAuthorize("hasRole('$ROLE_ADMIN')")
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  fun patchCustomSeriesThumbnail(
+    @AuthenticationPrincipal principal: KomgaPrincipal,
+    @PathVariable(name = "seriesId") seriesId: String,
+    @RequestParam("image") image: MultipartFile
+  ) {
+    val series = seriesRepository.findByIdOrNull(seriesId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    try {
+      ImageIO.read(image.inputStream).toString()
+      seriesLifecycle.addCustomThumbnailForSeries(series, image.bytes)
+    } catch (e: Exception) {
+      when (e) {
+        is IOException,
+        is IllegalArgumentException,
+        is NullPointerException -> throw ResponseStatusException(HttpStatus.UNSUPPORTED_MEDIA_TYPE)
+        is DataAccessException -> throw ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE)
+        else -> throw ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR)
+      }
+    }
+  }
+
+  @DeleteMapping("{seriesId}/thumbnail")
+  @PreAuthorize("hasRole('$ROLE_ADMIN')")
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  fun deleteCustomSeriesThumbnail(
+    @AuthenticationPrincipal principal: KomgaPrincipal,
+    @PathVariable(name = "seriesId") seriesId: String,
+  ) {
+    val series = seriesRepository.findByIdOrNull(seriesId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    try {
+      seriesLifecycle.deleteCustomThumbnailForSeries(series)
+    } catch (e: Exception) {
+      when (e) {
```

Comment:
That seems drastic. Any uncaught exception will always return an internal server error anyway. Probably not needed at all unless that's an exception our own code could throw.